src/main/java/com/example/restapp/dao/AbstractDAO.java

```diff
@@ -4,13 +4,13 @@
 
 public abstract class AbstractDAO {
 
-	protected BeanSession session;
-
-	public void setBeanSession(BeanSession session) {
+	private final BeanSession session;
+	
+	public AbstractDAO(BeanSession session) {
```

Comment:
I don't like this.. It forces the subclass to create the same constructor and call super(session). The injection by setter doesn't work anymore?